lib/ArangoDBClient/ConnectionOptions.php

```diff
@@ -422,6 +546,98 @@ private function validate()
         UpdatePolicy::validate($this->_values[self::OPTION_REPLACE_POLICY]);
         UpdatePolicy::validate($this->_values[self::OPTION_DELETE_POLICY]);
     }
+
+
+    /**
+     * load and merge connection options from optional Memcached cache into
+     * ihe current settings
+     *
+     * @return void
+     */
+    private function loadOptionsFromCache() 
+    {
+        $cache = $this->getEndpointsCache();
+
+        if ($cache === null) {
+            return;
+        }
+
+        $endpoints = $cache->get($this->_values[self::OPTION_MEMCACHED_ENDPOINTS_KEY]);
+        if ($endpoints) {
+            $this->_values[self::OPTION_ENDPOINT] = $endpoints;
+        }
+    }
+    
+    /**
+     * store the updated options in the optional Memcached cache
+     *
+     * @return void
+     */
+    private function storeOptionsInCache() 
+    {
+        $endpoints = $this->_values[self::OPTION_ENDPOINT];
+        $numberOfEndpoints = count($endpoints);
```

Comment:
@jsteemann If I'm not mistaken, `$this->_values[self::OPTION_ENDPOINT]`, may be a string here. While this returns 1, it's a little misleading and will trigger warnings in PHP >= 7.2.

Maybe force this to an array in the constructor to avoid `if`s everywhere?